MvvmCross-AndroidSupport/MvvmCross.Droid.Support.V7.RecyclerView/MvxRecyclerAdapter.cs

```diff
@@ -241,7 +243,15 @@ protected virtual void SetItemsSource(IEnumerable value)
 
         protected virtual void OnItemsSourceCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
         {
-            NotifyDataSetChanged(e);
+            if (Looper.MainLooper == Looper.MyLooper())
```

Comment:
Isn't there a Mvx method that checks the thread and runs it on main if necessary?

Reply:
There is `MvxAndroidMainThreadDispatcher`. However this class can't inherit from `MvxMainThreadDispatchingObject`. So this is a good solution.